tools/offline_log_viewer/kvstore.py

```diff
@@ -349,6 +351,10 @@ def decode(self):
             s = Segment(path)
             for batch in s:
                 for r in batch:
+                    offset = batch.header.base_offset + r.offset_delta
+                    if snapshot_offset is not None and offset < snapshot_offset:
```

Comment:
hmm, would offset be inclusive here: "all messages from before up to including the snapshot offset"?

Reply:
you are correct, I was thinking of next_offset, but the save_snapshot path in kvstore already subtracts 1.  Fixed.